int32 batteryCharge
int32 batteryCapacity
